src/events/attributes.rs

```diff
@@ -32,6 +35,93 @@ pub struct Attribute<'a> {
 }
 
 impl<'a> Attribute<'a> {
+    /// Returns the attribute value normalized as per the XML specification.
+    ///
+    /// https://www.w3.org/TR/xml/#AVNormalize
+    ///
+    /// Do not use this method with HTML attributes.
+    ///
+    /// Escape sequences such as `&gt;` are replaced with their unescaped equivalents such as `>`
+    /// and the characters \t, \r, \n are replaced with whitespace characters.
+    ///
+    /// This will allocate unless the raw attribute value does not require normalization.
+    ///
+    /// See also [`normalized_value_with_custom_entities()`](#method.normalized_value_with_custom_entities)
+    pub fn normalized_value(&'a self) -> Result<Cow<'a, [u8]>, EscapeError> {
```

Comment:
I think, it will be valuable to add examples here. Actually, probably you can convert your new tests to doc examples and that will be enough

Reply:
Actually, we should add a decoder parameter to that (and similar) functions and decode first before normalization.

Maybe also try another approach: introduce a new type of always UTF-8 encoded attributes and add a `Attribute::decode(&self) -> Result<Utf8Attribute>` function.

Instead of completely new type we could try to use `const bool` generic parameter (stable since 1.51, current MSRV is 1.41.1, from `memchr`)